packages/modelserver-client/src/model-server-client-v2.ts

```diff
@@ -0,0 +1,338 @@
+/*********************************************************************************
+ * Copyright (c) 2022 STMicroelectronics and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0, or the MIT License which is
+ * available at https://opensource.org/licenses/MIT.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR MIT
+ *********************************************************************************/
+import axios, { AxiosError, AxiosInstance, AxiosRequestConfig, AxiosResponse } from 'axios';
+import { Operation } from 'fast-json-patch';
+import WebSocket from 'isomorphic-ws';
+
+import { ModelServerError, ServerConfiguration, SubscriptionOptions } from './model-server-client-api-v1';
+import { Format, ModelServerClientApiV2, ModelUpdateResult } from './model-server-client-api-v2';
+import { MessageDataMapper, Model, ModelServerMessage } from './model-server-message';
+import { ModelServerPaths } from './model-server-paths';
+import { ModelServerCommand } from './model/command-model';
+import { Diagnostic } from './model/diagnostic';
+import { SubscriptionListener } from './subscription-listener';
+import { AnyObject, asObject, asString, asType, encodeRequestBody, TypeGuard } from './utils/type-util';
+
+/**
+ * A client to interact with a model server.
+ */
+export class ModelServerClientV2 implements ModelServerClientApiV2 {
```

Comment:
I saw on the server side that there is support for transactions. Should that be reflected in the client as well?

Reply:
Copying the reply from the other PR:

> It's used by a new project: https://github.com/eclipse-emfcloud/modelserver-node
> 
> We decided to exclude it from the Javascript/Java client API at the moment; it's considered an implementation detail, to > be used through the modelserver-node APIs (At the moment. But maybe it could be properly opened in the future).

Reply:
I understand, so it is more internal API that is not yet supposed to be "public". In that case I am fine with not exposing it for now. However, usually I am not a fan of hiding functionality that is obviously there so it might make more sense to explicitly mark it internal through comments or a second interface. But this should not prevent this PR from being merged.